lib/core.js

```diff
@@ -555,6 +555,12 @@ Archiver.prototype.append = function(source, data) {
  * @return {this}
  */
 Archiver.prototype.bulk = function(mappings) {
+  if (typeof process !== 'undefined' && typeof process.emitWarning !== 'undefined') {
+    process.emitWarning('Archiver.bulk() deprecated since 0.21.0', 'DeprecationWarning');
+  } else if (typeof console !== 'undefined' && typeof console.warn !== 'undefined') {
+    console.warn('Archiver.bulk() deprecated since 0.21.0');
+  }
```

Comment:
+1 test of console is overkill, also you might want to declare a variable for the error string to avoid repeating it.

Then 👍